pkg/apis/kubermatic/v1/common.go

```diff
@@ -18,7 +18,8 @@ package v1
 
 // +kubebuilder:validation:Enum=NodePort;LoadBalancer;Tunneling
 
-// ExposeStrategy is the strategy to expose the cluster with.
+// ExposeStrategy is the strategy used to expose a cluster control plane.
+// Possible values are `NodePort`, `LoadBalancer` or `Tunneling` (requires a feature gate).
```

Comment:
I am slightly worried that comments like these will quickly get out-of-sync with the surrounding code. Do you have a solution to keep them in-sync?

Reply:
I agree this is a problem, I'm not sure how to fix it though. IMHO, our CRDs need to expose this information, as it will otherwise leave users skimming through documentation until they find the valid values (if that is documented in another place at all). We could maybe enforce through validation a particular format on comments for enum fields so that they have to list the possible values, but I'm not entirely sure that is the way to go (or how to do it reliably).

Reply:
Well, at least we share the pain. So it's 4 eyes in the future to keep those comments up-to-date.

Reply:
@xrstf if it's of any comfort to you, adding new values to the CRD needs updating the `+kubebuilder:validation` annotation anyway, I'm banking on people seeing that it needs updating two lines down as well.